feature/tabsflow/moreflow/donate/src/main/java/com/epmedu/animeal/tabs/more/donate/presentation/viewmodel/DonateViewModel.kt

```diff
@@ -0,0 +1,45 @@
+package com.epmedu.animeal.tabs.more.donate.presentation.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.epmedu.animeal.common.presentation.viewmodel.delegate.DefaultStateDelegate
+import com.epmedu.animeal.common.presentation.viewmodel.delegate.StateDelegate
+import com.epmedu.animeal.tabs.more.donate.domain.DonateRepository
+import com.epmedu.animeal.tabs.more.donate.presentation.DonateScreenEvent
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class DonateViewModel @Inject constructor(private val donateRepository: DonateRepository) :
+    ViewModel(),
+    StateDelegate<DonateState> by DefaultStateDelegate(initialState = DonateState()) {
+
+    init {
+        viewModelScope.launch {
+            donateRepository.getDonateInformation().collect {
+                updateState { copy(donationInformation = it) }
+            }
+        }
+    }
+
+    fun handleEvent(event: DonateScreenEvent) {
+        when (event) {
+            is DonateScreenEvent.DonateNumberClicked -> {
+                updateState {
+                    copy(donationNumberToCopy = event.number)
+                }
+            }
+            is DonateScreenEvent.BackClicked -> {
+                updateState {
+                    copy(popBackstack = true)
+                }
+            }
+            is DonateScreenEvent.NumberIsCopied -> {
+                updateState {
+                    copy(donationNumberToCopy = null)
```

Comment:
You should use `EventDelegate` mechanism for cases that are one-time events, such as copy and navigation. This way you won't need `NumberIsCopied` event and code will be more readable. For example, you can check `FinishProfileViewModel` or any other viewmodel that uses the event delegate

Reply:
I was thinking hard about this comment, and re-read documentation on the topic :) 
So, regarding back navigation, I've removed it from events, so now it's handled on the screen itself (like we have on several other screens)
But regarding the EventDelegate I can't agree. 
Using one-off events is considered a bad practice, so I would say that the approach with updating state is quite OK 

Please check 
https://developer.android.com/topic/architecture/ui-layer/events
https://medium.com/androiddevelopers/viewmodel-one-off-event-antipatterns-16a1da869b95